### PR TITLE
Update CodeQL model editor info for revised UI

### PR DESCRIPTION
**Pull request by felicitymay**
This pull request updates the information on how to use the CodeQL model editor to reflect upcoming changes to the extension.

For more information about the changes, see the linked internal tracking issue.

### CODE REVIEW

File: docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst
Comment (felicitymay) [inline docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst:72]: @jf205 - should we consider updating this to mention that, by default, CodeQL only tracks dataflow from sources that are defined as "remote", and link out to information on also tracking dataflow from "local" sources? Or is that something for the future?

File: docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst
Comment (subatoi) [inline docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst:31]:
```suggestion
#. In the left side panel, expand the "CodeQL method modeling" section and click **Start modeling** to display the model editor. Alternatively, use the command palette to run the “CodeQL: Open Model Editor (Beta)” command.
```

File: docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst
Comment (subatoi) [inline docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst:69]:
```suggestion
#. When you have determined how to model your use of the method, you can define the **Model type** in the "CodeQL method modeling" tab of the CodeQL extension. This change is automatically reflected in the main model editor.
```

Not 100% sure this edit makes sense but I think the sentence might need to be broken up?

**Review by subatoi — DISMISSED** [review]
From a content POV this looks good to me; I'm not very knowledgeable about the subject matter but I'm happy to take another look when an SME has reviewed, if useful.

File: docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst
Comment (jf205) [reply docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst:132]: @starcke I think you might be able to explain this better than me. It is that a method is callable in different ways that means we want multiple classifications? I'm not sure that sounds right (but I could be wrong).

Comment (starcke) [reply docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst:132]: Each classification (or model) refers to a single flow (e.g. parameter 1 might be a sql-injection sink). Each method might have multiple flows. For example, consider a method that executes a query and returns it:
```
String runQuery(String query)
```
Here `q` is likely to be a sink for SQL injection. But at the same time because the query is also returned it is also a flow summery where taint can flow from the first parameter to the return value. This could potentially result in other vulnerabilities (e.g. a log injection vulnerability if the query is later logged).

Comment (starcke) [reply docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst:132]: The reason we need multiple classifications/models is to capture all the possible flows for a given method.

Comment (felicitymay) [reply docs/codeql/codeql-for-visual-studio-code/using-the-codeql-model-editor.rst:132]: Thanks for this discussion. I've updated the title and introduction to this section accordingly.

**Comment by felicitymay** [comment]
Thanks for the feedback @jf205 and @starcke ✨ 

I think I've addressed this and I'm now going to rebase the commits on `main` and retarget the pull request to reflect the new target ship date.

**Comment by felicitymay** [comment]
Apologies for the noise. I've just rebased this to retarget it against `main`, however, I have
one extra commit that doesn't really belong in this PR. It looks as if we may be missing a mergeback from the last release candidate branch to `main`.

**Comment by aeisenberg** [comment]
For this PR, since you are already rebasing it, the simplest approach would be to rewrite the history and remove any commits that you don't want to be included.

We can mergeback the other missing commits separately.

**Comment by felicitymay** [comment]
Many thanks @aeisenberg ✨ 

We just did a merge back and this PR should now contain only the intended changes. As a bonus, the next CodeQL CLI release won't revert the new combined change log 🙂

**Review by jf205 — APPROVED** [review]
Thank you for the updates @felicitymay. LGTM ⚡

**Comment by felicitymay** [comment]
@jf205 - thanks for the approval.

Should I wait for an approval from someone on the @github/codeql-vscode-reviewers team before merging? I'd like to get this merged before we create the GHES release branch next week.

**Comment by jf205** [comment]
>Should I wait for an approval from someone on the `@github/codeql-vscode-reviewers` team before merging?

I don't think so!

**Pull request merged**
